### PR TITLE
Reorganize "advanced" docs into "Developer Guide" 

### PR DESCRIPTION
**Pull request by ibgreen**
#### Background
For #1513 - Change "Advanced" to "Developer Guide"
#### Change List
- document index and some pages.

### CODE REVIEW

File: docs/get-started/using-standalone.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+# Using deck.gl without React (Experimental)
```
Comment (1chandu) [inline docs/get-started/using-standalone.md]: Nit: This heading is same as following sub-heading, may be change this to `Using deck.gl standalone` ?

**Comment by Pessimistress** [comment]
Our top-level categories are still confusing. My suggestions:
- Fold `custom layer guide` into a subsection of `developer guide`, and also make it clear that any future deep-dive articles must go into `developer guide` instead of a top-level category.
- Split `deckgl-test-utils` into articles and api references. Articles go into a subsection of `developer guide` called `testing`. API reference go into a top-level `deckgl-test-utils API`
- Rename the current `API-reference` to `deck.gl API` and redirect old links here

**Comment by ibgreen** [comment]
Updated per comments

File: docs/get-started/using-standalone.md (hunk 1 above)
Comment (ibgreen) [reply docs/get-started/using-standalone.md]: Good catch. Updated second heading to say "Example Code"

**Pull request merged**